src/class-wc-gateway-btcpay.php

```diff
@@ -973,14 +984,9 @@ public function ipn_callback()
                 $this->log('    [Info] Order ID is: ' . $order_id);
             }
 
-        //this is for the basic and advanced woocommerce order numbering plugins
-        //if we need to apply other filters, just add them in place of the this one
-            $order_id = apply_filters('woocommerce_order_id_from_number', $order_id);
```

Comment:
With above described bug it gets interesting here, the $order_number was passed here as $order_id and altered by filters leading to completely different results than intended? 😮

Reply:
Yep, that's been confusing and the source of errors when using a custom order plugin.